### PR TITLE
fixes ssh for servers with PubkeyAuthentication only

### PR DESCRIPTION
**Pull request by mgogoulos**
https://issues.apache.org/jira/browse/LIBCLOUD-461

### CODE REVIEW

File: libcloud/compute/ssh.py [hunk 1]
```diff
@@ -173,11 +173,8 @@ def connect(self):
 
         if self.password:
             conninfo['password'] = self.password
-        elif self.key:
+        if self.key:
             conninfo['key_filename'] = self.key
-        else:
-            conninfo['allow_agent'] = True
```
Comment (Kami) [inline libcloud/compute/ssh.py:179]: For explicitness sake and backward compatibility reasons I would prefer to still specify those two options.

**Comment by Kami** [comment]
I needed paramiko docs and ticket description to see what happens if both arguments (multiple authentication methods) are specified.

It appears paramiko just tries different authentication methods in a pre-defined order of priority so this change looks good.

**Comment by Kami** [comment]
Existing tests are failing btw - https://travis-ci.org/apache/libcloud/builds/15403675.

Those tests need to be fixed / updated. It would also be good to add a new test case which checks what happens when both arguments are specified.

**Comment by Kami** [comment]
I went ahead fixed the failing tests and made the `look_for_keys` and `allow_agent` change described above and merged this patch into trunk.

Thanks!

**Comment by mgogoulos** [comment]
Nice, thanks for adding the tests! Shame for me for postponing for such a long time!

**Pull request closed without merging**